src/factors/Inertial/IMUDeltaFactor.jl

```diff
@@ -0,0 +1,467 @@
+         v[2],
+        -v[1],
+            0
+    )
+end
+
+struct IMUDeltaManifold <: AbstractManifold{ℝ} end
+
+# TODO maybe define the manifold like this
+# M = ProductManifold(TranslationGroup(3), TranslationGroup(3), SpecialOrthogonal(3), Euclidean(1))
+# M = ProductManifold(SpecialOrthogonal(3), TranslationGroup(3), TranslationGroup(3), Euclidean(1))
+# G = GroupManifold(M, MultiplicationOperation())
+
+#TODO 
+# also see SE₂(3) 
+
+const IMUDeltaGroup = GroupManifold{ℝ, IMUDeltaManifold, MultiplicationOperation}
+
+IMUDeltaGroup() = GroupManifold(IMUDeltaManifold(), MultiplicationOperation())
+
+Manifolds.manifold_dimension(::IMUDeltaManifold) = 9
+
+# Affine representation 
+# Δ = [ΔR Δv Δp;
+#      0   1 Δt;
+#      0   0  1] ⊂ \R^5x5
+
+#  ArrayPartition representation (TODO maybe swop order to [Δp; Δv; ΔR; Δt])
+# Δ = [ΔR; Δv; Δp; Δt] 
+
+# TODO is SemidirectProductGroup possible, Translation(3), Translation(3), SO(3), WhatAction?, Δt?
+
+function Manifolds.identity_element(M::IMUDeltaGroup) # was #SMatrix{5,5,Float64}(I)
+    ArrayPartition(
+        SMatrix{3,3,Float64}(I), # ΔR
+        @SVector(zeros(3)),      # Δv
+        @SVector(zeros(3)),      # Δp
+        0.0                      # Δt
+    )
+end
+
+DFG.getPointIdentity(M::IMUDeltaGroup) = identity_element(M)
+
+function Manifolds.affine_matrix(G::IMUDeltaGroup, p::ArrayPartition{T}) where T<:Real
+    return SMatrix{5,5,T}([
+        p.x[1]  p.x[2]  p.x[3];
+        0 0 0        1  p.x[4];
+        0 0 0        0       1
+    ])
+end
+
+function Manifolds.inv(M::IMUDeltaGroup, p)
+    ΔR = p.x[1]
+    Δv = p.x[2]
+    Δp = p.x[3]
+    Δt = p.x[4]
+
+    return ArrayPartition(
+         ΔR',   
+        -ΔR'*Δv,
+        -ΔR'*(Δp - Δv*Δt),
+        -Δt
+    )
+end
+
+function Manifolds.compose(M::IMUDeltaGroup, p, q)
+    ΔR = p.x[1]
+    Δv = p.x[2]
+    Δp = p.x[3]
+    Δt = p.x[4]
+
+    δR = q.x[1]
+    δv = q.x[2]
+    δp = q.x[3]
+    δt = q.x[4]
+
+    return ArrayPartition(
+        ΔR * δR,
+        Δv + ΔR * δv,
+        Δp + Δv * δt + ΔR * δp,
+        Δt + δt
+    )
+end
+
+function Manifolds.hat(M::IMUDeltaGroup, Xⁱ::SVector{10, T}) where T<:Real
+    return ArrayPartition(
+        skew(Xⁱ[SA[7:9...]]), # θ ωΔt
+        Xⁱ[SA[1:3...]],       # ρ vΔt
+        Xⁱ[SA[4:6...]],       # ν aΔt
+        Xⁱ[10],               # Δt
+    )
+end
+
+function Manifolds.vee(M::IMUDeltaGroup, X::ArrayPartition{T}) where T<:Real
+    return SVector{10,T}(
+        X.x[2]...,   # ρ vΔt 1:3
+        X.x[3]...,   # ν aΔt 4:6
+        X.x[1][3,2], # θ⃗ₓ[3,2] 7
+        X.x[1][1,3], # θ⃗ₓ[1,3] 8 
+        X.x[1][2,1], # θ⃗ₓ[2,1] 9
+        X.x[4], # Δt 10
+    )
+end
+
+# small angle approx:
+# Q ≈ I
+# P ≈ 1/2 * I
+
+function _Q(θ⃗)
+    T = eltype(θ⃗)
+    θ = norm(θ⃗)
+    if θ ≈ 0
+        return SMatrix{3,3,T}(I)
+    else
+        u = θ⃗/θ
+        sθ, cθ = sincos(θ)
+        uₓ = skew(u)
+        #TODO possible mistake in source here with ^2 --- (θ - sθ)/θ^2 vs (θ - sθ)/θ
```

Comment:
That's easy to check by comparing to matrix exponential of the entire 5x5 matrix.